definition_tooling/converter/tests/test_converter.py

```diff
@@ -10,10 +10,18 @@
     convert_data_product_definitions,
 )
 
+# This needs to be a relative path to work correctly on Windows!
+# On Windows, if the path is not relative, in other words it's absolute, we will end up
+# with some C-Input/C-Output instead of the class names for the Request/Response classes
+# in the final OpenAPI spec files.
+# Below is a link to the issue on GitHub
+# https://github.com/pydantic/pydantic/issues/7860
+definitions_dir = Path(__file__).parent.joinpath("data").relative_to(Path.cwd())
```

Comment:
We could actually get rid of this whole workaround if we ensure the module is loaded with a sane name.

The code in `convert_data_product_definitions` right now looks like this:
```python
spec = importlib.util.spec_from_file_location(name=str(p), location=str(p))
        if not spec.loader:
            raise RuntimeError(f"Failed to import {p} module")
        try:
            module = spec.loader.load_module(str(p))
        except ValidationError as e:
            should_fail_hook = True
            print(styled_error("Validation error", p))
            print(e)
            continue
```

The problem is that the `name` attribute to `importlib.util.spec_from_file_location` gets the `C:\...` on Windows. But a python module should not really have a colon in the name. When we were earlier debugging this, we tried to change it and it failed with some errors to import it if we tried to change it.

The recommended way to do the import is described at https://docs.python.org/3/library/importlib.html#importing-a-source-file-directly and looks pretty much like this:

```python
spec = importlib.util.spec_from_file_location(module_name, file_path)
module = importlib.util.module_from_spec(spec)
sys.modules[module_name] = module
spec.loader.exec_module(module)
```
With that approach you will be able to set some sane name for the module name. I think we should actually try to get the names converted to somewhat sane ones, i.e. replace the slash (or backslash on Windows) with a dot and make the module name start from src, so either `src.AirQuality.Current_v1.0` or `AirQuality.Current_v1.0`. Of course the `.` is also reserved, so that should likely also be transliterated to for example an underscore. I'm not 100% sure if we should try to convert it to snake case as well or not. For now, I think even the ones I first suggested could do. Later when we want to support definitions that can include/import things from other definitions we might want to consider this in more detail.

Also since the definitions do take up a fair bit of memory, I'd suggest skipping the row that adds them to sys.modules (i.e. skip `sys.modules[module_name] = module`). I believe it won't be needed for this (would likely however be needed for those that we later want to be able to use in other definitions, i.e. the shared ones).

If this feels like a too big undertaking as part of this task, feel free to file it as a new task request and/or issue in this repo (and a task request linking to it).